tm2/pkg/sdk/bank/keeper.go

```diff
@@ -21,24 +22,84 @@ type BankKeeperI interface {
 	SubtractCoins(ctx sdk.Context, addr crypto.Address, amt std.Coins) (std.Coins, error)
 	AddCoins(ctx sdk.Context, addr crypto.Address, amt std.Coins) (std.Coins, error)
 	SetCoins(ctx sdk.Context, addr crypto.Address, amt std.Coins) error
+
+	InitGenesis(ctx sdk.Context, data GenesisState)
+	GetParams(ctx sdk.Context) Params
 }
 
-var _ BankKeeperI = BankKeeper{}
+var _ BankKeeperI = &BankKeeper{}
 
 // BankKeeper only allows transfers between accounts without the possibility of
 // creating coins. It implements the BankKeeperI interface.
 type BankKeeper struct {
 	ViewKeeper
 
 	acck auth.AccountKeeper
+	// The keeper used to store parameters
+	paramk           params.ParamsKeeper
+	params           Params
+	restrictedDenoms map[string]struct{}
```

Comment:
Is there a way we can have this not be part of the `BankKeeper` directly?
I can definitely see the use case of it, but it may be something that becomes very outdated quickly, and we'd have to drag it around in all future implementations to maintain backwards compatibility

Reply:
Not sure if this is a temporary solution for two reasons; however, we can have a separate discussion on this:

- There was a requirement to lock or unlock transfers for accounts based on certain conditions being met.
- To address malicious use of issued native tokens, GovDAO has the capability to control them.